File: lib/dingen_handler_random.erl
-module(dingen_handler_random).

-export([init/3, handle/2, terminate/3]).



init(_Type, Req, _Opts) ->
  {ok, Req, no_state}.


handle(Req, State) ->
  Headers = [{"content-type", "text/html"}],
  Random = gen_server:call(dingen_dispenser, random),
  Body = [
    "<title>Hello ", Random, "!</title>",
    "<p>Hello ", Random, "!</p>"
  ],
  {ok, Req2} = cowboy_req:reply(200, Headers, Body, Req),
  {ok, Req2, State}.


terminate(_Reason, _Req, _State) ->
  ok.

File: lib/dingen_handler_random.ex
defmodule Dingen.Handler.Random do
  def init(_type, req, _opts) do
    {:ok, req, :no_state}
  end

  def handle(req, state) do
    headers = [{"content-type", "text/html"}]
    random = GenServer.call(:dingen_dispenser, :random)
    body = "
      <title>Hello #{random}!</title>
      <p>Hello #{random}!</p>
    "
    {:ok, req} = :cowboy_req.reply(200, headers, body, req)
    {:ok, req, state}
  end

  def terminate(_reason, _req, _state) do
    :ok
  end
end
